internal/auth/yaml_storage.go

```diff
@@ -8,18 +8,30 @@ import (
 	"io/fs"
 	"os"
 
+	"github.com/charmbracelet/log"
 	"github.com/inovex/CalendarSync/internal/config"
 	"gopkg.in/yaml.v3"
 )
 
 type YamlStorage struct {
 	StoragePath          string
 	StorageEncryptionKey string
+	// Holds the decrypted CalendarAuth Config in memory, so the file does not have to be read multiple times
+	DecryptedAuth []CalendarAuth
```

Comment:
(Nitpick) I'd slightly prefer `CachedAuth`, although I'm also ok with keeping the current name.

Reply:
CachedAuth is a way better name imo. Renamed